### PR TITLE
Pet location

### PR DESCRIPTION
**Pull request by bsimps01**
# Pull Request Template

## Description

-Was able to successfully apply pet location based on zip code of pet
-Applied working buttons to menu view to go in between working views
-Included different views where other tasks will be completed

Fixes # 33

## Front end, Additional Feature

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

-Used simulator for changing different features

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: Petulia/Constants/Constants.swift [hunk 1]
```diff
@@ -77,5 +77,7 @@ extension AnimalKeys {
   enum Contact {
     static let email = "email"
     static let phone = "phone"
+    static let address = "address"
   }
+
```
Comment (SamuelFolledo) [inline Petulia/Constants/Constants.swift]: remove this extra space

File: Petulia/Models/PetResponse.swift [hunk 2]
```diff
@@ -181,22 +181,27 @@ extension Attributes {
...
-  init(email: String, phone: String) {
+
+  init(email: String, phone: String, address: Address) {
     self.email = email
     self.phone = phone
+    self.address = address
   }
   
-  var asDictionary: [String: String] {
+
```
Comment (SamuelFolledo) [inline Petulia/Models/PetResponse.swift]: remove extra whitespace

File: Petulia/Views/Authentication/LoginView.swift [hunk 3]
```diff
@@ -9,7 +9,7 @@
 import SwiftUI
 
 struct LoginView: View {
-  
+  //@State var viewRouter: ViewRouter
```
Comment (SamuelFolledo) [inline Petulia/Views/Authentication/LoginView.swift]: remove this

File: Petulia/Views/Authentication/LoginView.swift [hunk 4]
```diff
@@ -61,6 +61,7 @@ extension LoginView {
   }
   
   func goToHomePage() {
+
```
Comment (SamuelFolledo) [inline Petulia/Views/Authentication/LoginView.swift:63]: remove extra whitespace

File: Petulia/Views/DetailUI/PetDetailView.swift [hunk 5]
```diff
@@ -9,21 +9,26 @@
 import SwiftUI
 import Foundation
 import MessageUI
+import MapKit
 
 struct PetDetailView: View {
   
   var viewModel: PetDetailViewModel
+  var addressLocation: Address?
+  //var address1: address
```
Comment (SamuelFolledo) [inline Petulia/Views/DetailUI/PetDetailView.swift]: remove this line

File: Petulia/Views/DetailUI/PetDetailView.swift [hunk 6]
```diff
@@ -9,21 +9,26 @@
...
   @EnvironmentObject var favorites: FavoriteController
   @EnvironmentObject var theme: ThemeManager // to pass to sheets.
   @AppStorage(Keys.isDark) var isDark = false
   
+
```
Comment (SamuelFolledo) [inline Petulia/Views/DetailUI/PetDetailView.swift:21]: remove extra whitespace

File: Petulia/Views/DetailUI/PetDetailView.swift [hunk 7]
```diff
@@ -9,21 +9,26 @@
...
   
+  
   @State private var showAdoptionForm: Bool = false
   @State private var zoomingImage = false
   @State var result: Result<MFMailComposeResult, Error>? = nil
   @State private var showingMailView = false
   @State private var showingPhoneAlert = false
   @State private var showingEmailAlert = false
   
+
```
Comment (SamuelFolledo) [inline Petulia/Views/DetailUI/PetDetailView.swift:29]: remove extra whitespace

File: Petulia/Views/DetailUI/PetDetailView.swift [hunk 8]
```diff
@@ -36,6 +41,12 @@ struct PetDetailView: View {
             Spacer()
           }
           .padding()
+          if let postCode = viewModel.contact?.address?.postcode{
+            MapView(address: postCode)
+              .frame(width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height/6, alignment: .center)
+              .padding()
+          }
+
```
Comment (SamuelFolledo) [inline Petulia/Views/DetailUI/PetDetailView.swift:47]: remove extra whitespace

File: Petulia/Views/Home/HomeView.swift [hunk 9]
```diff
@@ -54,31 +44,9 @@ struct HomeView: View {
...
-            Image(systemName: "multiply").font(.body).foregroundColor(.white)
-              .imageScale(.large)
-          } else{
-            Image(systemName: "line.horizontal.3")
-              .imageScale(.large)
-          }
-        })
-      ), trailing: HStack { settingsControlView() })
     }
+
```
Comment (SamuelFolledo) [inline Petulia/Views/Home/HomeView.swift:53]: remove extra whitespace

File: Petulia/Views/MapView/MapView.swift [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  var locationManager = CLLocationManager()
+  @State var coordinate = CLLocationCoordinate2D()
+  //@Binding var pins: [Pin]
+  
+  func makeUIView(context: UIViewRepresentableContext<MapView>) -> MKMapView {
+    setupManager()
+    let mapView = MKMapView(frame: .zero)
+     //mapView.userTrackingMode = .follow
+    return mapView
+
```
Comment (SamuelFolledo) [inline Petulia/Views/MapView/MapView.swift:24]: remove extra whitespace

File: Petulia/Views/MapView/MapView.swift [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  func makeUIView(context: UIViewRepresentableContext<MapView>) -> MKMapView {
+    setupManager()
+    let mapView = MKMapView(frame: .zero)
+     //mapView.userTrackingMode = .follow
+    return mapView
+    
+  }
+  
+  func updateUIView(_ uiView: MKMapView, context: Context) {
+
```
Comment (SamuelFolledo) [inline Petulia/Views/MapView/MapView.swift]: remove extra whitespace and non documentation comments

File: Petulia/Views/MapView/MapView.swift [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+      }
+      coordinate = location
+      let span = MKCoordinateSpan(latitudeDelta: 0.5, longitudeDelta: 0.5)
+      let region = MKCoordinateRegion(center: coordinate, span: span)
+      uiView.setRegion(region, animated: true)
+      
+      let annotation = MKPointAnnotation()
+      annotation.coordinate = coordinate
+    }
+
```
Comment (SamuelFolledo) [inline Petulia/Views/MapView/MapView.swift]: remove extra whitespace

File: Petulia/Views/Menu/MenuView.swift [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+//
+//  Created by Benjamin Simpson on 4/29/21.
+//  Copyright © 2021 Johandre Delgado . All rights reserved.
+//
+
+import Foundation
+
+struct MenuView: View {
+  @EnvironmentObject var theme: ThemeManager
+  //@State var currentPage: Page = .homeView
```
Comment (SamuelFolledo) [inline Petulia/Views/Menu/MenuView.swift:13]: remove extra whitespace and non documentation comments

File: Petulia/Views/Menu/MenuView.swift [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+      }
+      .padding(.top, 20)
+      Spacer()
+    }
+    .padding()
+    .frame(maxWidth: UIScreen.main.bounds.width/2, alignment: .leading)
+    .background(theme.accentColor)
+    .edgesIgnoringSafeArea(.all)
+  }
+
```
Comment (SamuelFolledo) [inline Petulia/Views/Menu/MenuView.swift:62]: remove extra whitespace

File: Petulia/Views/MenuView/MenuView.swift [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+
+struct NavigationHamburgerMenu: View {
+  @EnvironmentObject var theme: ThemeManager
+  @StateObject var viewRouter: ViewRouter
+  @State var showMenu = false
+  @State private var showSettingsSheet = false
+  @AppStorage(Keys.savedPostcode) var postcode = ""
+  @EnvironmentObject var petDataController: PetDataController
+  
+    var body: some View{
```
Comment (SamuelFolledo) [inline Petulia/Views/MenuView/MenuView.swift]: fix tabbing

Comment (SamuelFolledo) [reply Petulia/Views/MenuView/MenuView.swift]: fix tabbing and remove unnecessary whitespace/comments

File: Petulia/Views/Settings/SettingsView.swift [hunk 16]
```diff
@@ -9,7 +9,7 @@
 import SwiftUI
 
 struct SettingsView: View {
-  @Environment(\.presentationMode) var presentationMode
+  //@Environment(\.presentationMode) var presentationMode
```
Comment (SamuelFolledo) [inline Petulia/Views/Settings/SettingsView.swift]: fix tabbing and remove unnecessary whitespace/comments

**Review by SamuelFolledo — CHANGES_REQUESTED** [review]
Clean your code. Fix tabbing and remove unnecessary whitespace/comments

**Review by bsimps01 — COMMENTED** [review]
Updated related changes and cleaned up files
